README.md

```diff
@@ -2,548 +2,282 @@
 
 # debug.rb
 
-This library provides debugging functionality to Ruby (MRI) 2.6 and later.
-
-This debug.rb is replacement of traditional lib/debug.rb standard library which is implemented by `set_trace_func`.
-New debug.rb has several advantages:
-
-* Fast: No performance penalty on non-stepping mode and non-breakpoints.
-* [Remote debugging](#remote-debugging): Support remote debugging natively.
-  * UNIX domain socket (UDS)
-  * TCP/IP
-  * Integration with rich debugger frontends
-
-     Frontend |  [Console](https://github.com/ruby/debug#invoke-as-a-remote-debuggee) | [VSCode](https://github.com/ruby/debug#vscode-integration) | [Chrome DevTool](#chrome-devtool-integration) |
+This library provides debugging functionality to Ruby (MRI) 2.6 and later. It has several advantages:
+
+- Fast: No performance penalty on non-stepping mode and non-breakpoints.
+- Native remote debugging support:
+  - [UNIX domain socket](/docs/remote_debugging.md#unix-domain-socket-uds) (UDS)
+  - [TCP/IP](/docs/remote_debugging.md#tcpip)
+  - Integration with rich debugger frontends
+     Frontend |  [Console](/docs/remote_debugging.md#debugger-console) | [VSCode](/docs/remote_debugging.md#vscode) | [Chrome DevTools](/docs/remote_debugging.md#chrome-devtool-integration) |
      ---|---|---|---|
      Connection | UDS, TCP/IP | UDS, TCP/IP | TCP/IP |
      Requirement | No | [vscode-rdbg](https://marketplace.visualstudio.com/items?itemName=KoichiSasada.vscode-rdbg) | Chrome |
-
-* Extensible: application can introduce debugging support with several ways:
-  * By `rdbg` command
-  * By loading libraries with `-r` command line option
-  * By calling Ruby's method explicitly
-* Misc
-  * Support threads (almost done) and ractors (TODO).
-  * Support suspending and entering to the console debugging with `Ctrl-C` at most of timing.
-  * Show parameters on backtrace command.
-  * Support recording & reply debugging.
+  - See the [remote debugging guide](/docs/remote_debugging.md) for details
+- Flexible: Users can use the debugger in multiple ways
+  - Through requiring files - like `require "debug"`
+  - Through the [`rdbg` executable](#the-rdbg-executable)
+  - Through Ruby APIs
+  - See [activate the debugger in your program](#activate-the-debugger-in-your-program) for details
+- Configurable:
+  - It provides many configurations and is configurable through the `~/.rdbgrc` file
+  - See the [configuration guide](/docs/configuration.md) for more information
```

Comment:
I think we can add a `docs/concurrency_support.md` and add a `Concurrency support:` here for information and multi-thread/process and Ractor support. But I'm not sure what'd be the content in it so I haven't done it.